src/jl_uv.c

```diff
@@ -94,12 +94,17 @@ DLLEXPORT void jl_get_uv_hooks()
 
 extern jl_module_t *jl_old_base_module;
 // Use:  JULIA_CB(hook, arg1, numberOfAdditionalArgs, arg2Type, arg2, ..., argNType, argN)
-#define JULIA_CB(hook,val, ...) \
+    // (!jl_old_base_module ? (
+        // jl_callback_call(JULIA_HOOK(hook),(jl_value_t*)val,__VA_ARGS__)
+    // ) : (
+#define JULIA_CB(hook,mod,val, ...) \
     (!jl_old_base_module ? ( \
-        jl_callback_call(JULIA_HOOK(hook),(jl_value_t*)val,__VA_ARGS__) \
+        jl_callback_call( \
+            JULIA_HOOK_((jl_module_t*)jl_get_binding(jl_base_module, mod)->value, hook), \
+            (jl_value_t*)val,__VA_ARGS__) \
     ) : ( \
         jl_callback_call( \
-            JULIA_HOOK_(jl_base_relative_to(((jl_datatype_t*)jl_typeof(val))->name->module), hook), \
+            JULIA_HOOK_((jl_module_t*)jl_get_binding(jl_base_relative_to(((jl_datatype_t*)jl_typeof(val))->name->module), mod)->value, hook), \
```

Comment:
since we are doing `jl_apply_generic` here anyways, the extra cost of a runtime lookup shouldn't be noticeable. so this can just be:
`JULIA_HOOK_( ((jl_datatype_t*)jl_typeof(val))->name->module, hook ), \ `

and `jl_get_uv_hooks()` can go away

(this code is due for a more intensive refactoring soon anyways, now that we have full cfunction support)